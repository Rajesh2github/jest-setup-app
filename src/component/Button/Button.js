import React from 'react';

export default function Button({ label, onClick, style }) {
  return (
    <div>
    <button style={{...style}}onClick={onClick}>{label}</button>
    </div>
  );
}
