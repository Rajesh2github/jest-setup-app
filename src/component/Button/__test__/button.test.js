import { render, screen, fireEvent } from "@testing-library/react";
import Button from "../Button";

// Snapshot Test
test("Button snapshot", () => {
  const { asFragment } = render(<Button label="ok" onClick={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

// Text rendering test
test("renders the button label", () => {
  render(<Button label="Click Me" onClick={() => {}} />);
  expect(screen.getByText("Click Me")).toBeInTheDocument();
});

// Click event test
test("calls onClick when clicked", () => {
  const handleClick = jest.fn();
  render(<Button label="Click Me" onClick={handleClick} />);

  fireEvent.click(screen.getByText("Click Me"));
  expect(handleClick).toHaveBeenCalledTimes(1);
});

// Accessibility test (role check)
test("button has role 'button'", () => {
  render(<Button label="Click Me" onClick={() => {}} />);
  expect(screen.getByRole("button")).toBeInTheDocument();
});
