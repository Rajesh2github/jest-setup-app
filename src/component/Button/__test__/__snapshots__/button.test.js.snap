// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      ok
    </button>
  </div>
</DocumentFragment>
`;
